fix: Chapter Master and Captains missing in various places

EttyKitty opened the pull request:
There was a bug with captain's and chapter master names not displaying on the management screen - that I fixed.
There was a bug with captains and chapter master not de-selecting on drop select - that I fixed as well.

There is [this](https://discord.com/channels/714022226810372107/1304818036381057075)
> Deploy marines to ground combat but the captains was never there.I tried unload a captain alone on the enemy planet,next turn the ground combat didn't happen.I also tried unload a entire command squad,captain is still missing in combat.

bug, that I can't fucking fix. I've spent hours trying to find that fucking array that skips CM and Captains that are at index 0.


I can't. I give up.

![image](https://github.com/user-attachments/assets/71642bd7-df89-4292-af0a-c8ed00b7e759)

EttyKitty committed:
fix: CM and captains display on management

EttyKitty committed:
God, give me strength to not have a stroke from these arrays of arrays

EttyKitty committed:
More arrays for Array God

OH296 commented on objects/obj_drop_select/Alarm_1.gml:
```diff
@@ -53,42 +53,45 @@ if (arright=false) then formation_current=formation_possible[1];
-                if (string_count("Bike",obj_ini.role[co][i])=0) or (attack=0){
-                    if (obj_ini.role[co][i]=obj_ini.role[100][11]) then l_mahreens+=1;
+var unit;
+for (var co = 1; co <= 10; co++) {
```
```suggestion
for (var co = 0; co <= 10; co++) {
```

OH296 commented on scripts/scr_management/scr_management.gml:
```diff
@@ -97,7 +97,7 @@ function scr_management(argument0) {
 	    nam[2]=role_names[Role.HONOUR_GUARD];
 
-	    for (var i = 1; i <= 200; i++) {
+	    for (var i = 0; i <= 200; i++) {
```
```suggestion
	    for (var i = 0; i <= array_length(obj_ini.name[0]); i++) {
```

OH296 commented on scripts/scr_management/scr_management.gml:
```diff
@@ -382,7 +382,7 @@ function scr_management(argument0) {
 			nam[23] = "Land Speeder";
 			nam[24] = "Whirlwind";
-	        for (var i=1;i<500;i++) {
+	        for (var i=0;i<500;i++) {
```
```suggestion
	        for (var i=0;i<array_length(obj_ini.name[0]);i++) {
```

OH296 commented:
i have solutions on my own branch of this,will make a pr to this branch

OH296 committed:
feat: Timer to limit combat turns (#100)

fix: Enemy rows phasing through (#100)
fix: Captains and CM not taking part in battle (#100)

OH296 reviewed (approved):
add suggestions then good to go

EttyKitty committed:
Apply suggestions from code review

Co-authored-by: Nelsonh <81228864+OH296@users.noreply.github.com>
